src/models/gun.rs

```diff
@@ -1,53 +1,71 @@
+const MAX_TEMP: f32 = 1.0;
+const MIN_TEMP: f32 = 0.0;
+const HEAT_PER_SHOT: f32 = 0.05;
+const NATURAL_COOL_DOWN_RATE: f32 = 0.4;
+const OVERHEAT_COOL_DOWN_RATE: f32 = 0.2;
+
+/// This is the player's gun: it slowly overheats with every shot so that the player can't just 
+/// spam shooting and ruin the gameplay
 pub struct Gun {
-    pub minimum_temperature: f32,
-    pub maximum_temperature: f32,
+    /// The current temperature of the gun - a percentage between 0 and 1
     pub temperature: f32,
-    heat_up_per_shot: f32,
-    natural_cool_down_rate: f32,
+    /// Whether or not the gun has overheated
     overheated: bool,
-    overheat_cool_down_rate: f32
 }
 
+/// We implement the Default trait for Gun because the Player struct has the Default trait, and so
+/// all of its fields require it oo
 impl Default for Gun {
     fn default() -> Gun { 
         Gun {
-            minimum_temperature: 0.0,
-            maximum_temperature: 1.0,
             temperature: 0.0,
-            heat_up_per_shot: 0.05,
-            natural_cool_down_rate: 0.2,
-            overheat_cool_down_rate: 0.4,
             overheated: false
         }
      }
 }
 
 impl Gun {
-    pub fn new() -> Gun { Gun::default() }
+    pub fn new() -> Gun {
+        Gun::default()
+    }
+
+    /// This is called every tick and slowly cools the gun down
     pub fn cool_down(&mut self, dt: f32){
         if self.overheated{
             self.overheat_cool_down(dt);
-        }else {
+        } else {
             self.natural_cool_down(dt);
         }
     }
+
+    /// Whenever the gun is fired it heats up
     pub fn heat_up(&mut self){
-        self.temperature = self.maximum_temperature.min(self.temperature + self.maximum_temperature * self.heat_up_per_shot);
-        if self.temperature == self.maximum_temperature {
+        self.temperature = MAX_TEMP.min(self.temperature + MAX_TEMP * HEAT_PER_SHOT);
+        if self.temperature == MAX_TEMP {
             self.overheated = true;
         }
     }
-    pub fn is_available(&mut self) -> bool { !self.overheated }
+
+    /// Check if the gun has overheated or not
+    pub fn is_available(&self) -> bool {
+        !self.overheated
+    }
+
+    /// Reset the gun's state back to its defaults
     pub fn reset(&mut self){
-        self.temperature = self.minimum_temperature;
+        self.temperature = MIN_TEMP;
         self.overheated = false;
     }
+
+    /// Cool down the gun naturally
     fn natural_cool_down(&mut self, dt: f32){
-        self.temperature = self.minimum_temperature.max(self.temperature - self.natural_cool_down_rate * dt);
+        self.temperature = MIN_TEMP.max(self.temperature - NATURAL_COOL_DOWN_RATE * dt);
     }
+
+    /// The gun cools down slower if it has overheated
```

Comment:
It cools down faster after an overheat!

Reply:
Oh, do we want it to cool down slower after an overheat or faster?

I made it slower after an overheat 'cause I thought it was more of a penalty to the player.

Reply:
It should cool down faster after an overheat. The punishment is not being able to shoot.